classes/migration/upgrade/v3_4_0/I8933_EventLogLocalized.php

```diff
@@ -0,0 +1,243 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_4_0/I8933_EventLogLocalized.php
+ *
+ * Copyright (c) 2023 Simon Fraser University
+ * Copyright (c) 2023 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I8933_EventLogLocalized.php
+ *
+ * @brief Adds a column to the event_log_settings table to store localized data such as a file name and drops setting_type column.
+ * In the event_log table allows null values for userId.
+ * Fixes the issue with duplicate event types and renames conflicting setting names
+ */
+
+namespace PKP\migration\upgrade\v3_4_0;
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+use PKP\install\DowngradeNotSupportedException;
+use PKP\log\event\PKPSubmissionEventLogEntry;
+use PKP\log\event\SubmissionFileEventLogEntry;
+use PKP\migration\Migration;
+
+abstract class I8933_EventLogLocalized extends Migration
+{
+    abstract protected function getContextTable(): string;
+
+    abstract protected function getContextIdColumn(): string;
+
+    /**
+     * Run the migration.
+     */
+    public function up(): void
+    {
+        Schema::table('event_log_settings', function (Blueprint $table) {
+            $table->string('locale', 14)->default('')->after('log_id');
+            $table->dropUnique('event_log_settings_unique');
+            $table->unique(['log_id', 'locale', 'setting_name'], 'event_log_settings_unique');
+            $table->string('setting_type', 6)->nullable()->change();
+        });
+
+        // Events can be triggered without a user, e.g., in schedule tasks
+        Schema::table('event_log', function (Blueprint $table) {
+            $table->dropForeign('event_log_user_id_foreign');
+            $table->dropIndex('event_log_user_id');
+            $table->bigInteger('user_id')->nullable()->change();
+            $table->foreign('user_id')->references('user_id')->on('users')->onDelete('cascade');
```

Comment:
Are you sure you want an `ON DELETE CASCADE` here? I would think `ON DELETE SET NULL` would be more appropriate -- the point is to be able to continue to log events when a user account doesn't exist, right?

Reply:
Correct me if I'm wrong: the only appropriate way to remove a user account is to merge it with another one. For this specific case we also combine their log entries. Thus, on delete cascade here would prevent data corruption in unexpected cases. If not, I can set to null on delete.

Forgot to mention, the idea behind making `user_id` nullable, is to allow to log events from the system, automated events or other cases when a specific user isn't involved.

Reply:
That makes sense to me, thanks, please go ahead. But please do add a `->comment` to the `user_id` column in `LogMigration` to explain what a null value means (we'll add more of these in future releases)!